src/tds/login.c

```diff
@@ -908,7 +911,7 @@ tds7_send_login(TDSSOCKET * tds, const TDSLOGIN * login)
 
 #define SET_FIELD_DSTR(field, dstr) do { \
 	data_fields[field].ptr = tds_dstr_cstr(&(dstr)); \
-	data_fields[field].len = tds_dstr_len(&(dstr)); \
+	data_fields[field].len = MIN(tds_dstr_len(&(dstr)), 128); \
```

Comment:
would be better to cap the converted string, not at the source.

Reply:
That's fair, but it looks like conversion here is via `tds_convert_stream`, which doesn't accept a length limit.  I suppose it might work to adjust `data_stream.size` directly.

Reply:
yes, done with also a test

Reply:
Great, thanks!  In general, sorry for not adding more tests -- our testing generally takes place via higher-level C++ wrappers.